toolkit/tools/imagegen/configuration/disk.go

```diff
@@ -21,13 +23,73 @@ type Disk struct {
 	RawBinaries        []RawBinary        `json:"RawBinaries"`
 }
 
+// checkOverlappingPartitions checks that start and end positions of the defined partitions don't overlap.
+func checkOverlappingePartitions(disk *Disk) (err error) {
+	partIntervals := [][]uint64{}
```

Comment:
Very small nit, but this could probably just be `[]*Disk`. No need to change, just took me a sec to wrap my head around the list of lists.